src/coprocessor/codec/mysql/duration.rs

```diff
@@ -244,6 +244,28 @@ impl Duration {
         self.dur = StdDuration::new(self.dur.as_secs(), nanos);
         Ok(self)
     }
+
+    pub fn checked_add(self, rhs: &Duration) -> Option<Duration> {
+        let add = match self.to_nanos().checked_add(rhs.to_nanos()) {
+            Some(result) => result,
+            None => return None,
+        };
+        match Duration::from_nanos(add, self.fsp().max(rhs.fsp()) as i8) {
+            Ok(res) => Some(res),
+            Err(_) => None,
+        }
+    }
+
+    pub fn checked_sub(self, rhs: &Duration) -> Option<Duration> {
```

Comment:
Could we use `&self` instead of `self`?

Reply:
yes, we can, but I think we should take ownership in add and subtract operation, like `checked_sub_signed` of Datetime<Tz> and `checked_add` of `i64`